zdigness/puttdle
# [TASK] Update prettier and vite dependencies

## Description

Prettier was implemented in [#14](https://github.com/zdigness/puttdle/pull/14) but has had issues with GitHub converting our standard of LF to CRLF.

Adding HMR support for vite dockerized client to allow seeing code changes when the container is already running

Lastly, Vite wasn't set as a dev dependency inside the client so docker-compose couldn't launch it.

## Motivation and Context

1. Enforce Git to keep LF line endings using [.gitattributes](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings)
2. Volume mount the client directory into the image and update the vite.config to include usePolling.
3. Add Vite dependencies

## How Has This Been Tested?

1. Go into root dir
2. Run docker-compose up
3. Change any frontend code and see changes applied

## Artifacts (if appropriate):

https://github.com/zdigness/puttdle/assets/60762808/5cfc90bf-118c-4688-9386-ad52fed17869
